src/main/java/com/clickhouse/kafka/connect/sink/state/provider/KeeperStateProvider.java

```diff
@@ -117,6 +123,11 @@ public void setStateRecord(StateRecord stateRecord) {
         long minOffset = stateRecord.getMinOffset();
         long maxOffset = stateRecord.getMaxOffset();
         String key = stateRecord.getTopicAndPartitionKey();
+
+        if (!(csc.getConnectStatePrefix() == null || csc.getConnectStatePrefix().isEmpty())) {
```

Comment:
can we flip the conditions inside? `(!())` gets confusing
`csc.getConnectStatePrefix() != null && !csc.getConnectStatePrefix().isEmpty()`

if theres a .isnotempty() or some other method that allows not using `!` here, that would be even better